packages/e2e-tests/specs/editor/blocks/gallery.test.js

```diff
@@ -52,6 +61,50 @@ describe( 'Gallery', () => {
 		expect( await getEditedPostContent() ).toMatch( regex );
 	} );
 
+	it( 'gallery caption can be edited', async () => {
+		const galleryCaption = 'Tested gallery caption';
+
+		await insertBlock( 'Gallery' );
+		const filename = await upload( '.wp-block-gallery input[type="file"]' );
+
+		await getUploadedFileFigure( filename );
+
+		await page.click( '.wp-block-gallery>.blocks-gallery-caption' );
+		await page.keyboard.type( galleryCaption );
+
+		expect( await getEditedPostContent() ).toMatch(
+			new RegExp( `<figcaption.*?>${ galleryCaption }</figcaption>` )
+		);
+	} );
+
+	it( "uploaded images' captions can be edited", async () => {
+		await insertBlock( 'Gallery' );
+		const filename = await upload( '.wp-block-gallery input[type="file"]' );
+
+		const figureElement = await getUploadedFileFigure( filename );
```

Comment:
Since you're uploading just one image here, the test can be really simplified by removing the `getUploadedFileFigure` function completely and replace it with:
```
const figureElement = await page.waitForSelector(
	'.blocks-gallery-item figure'
);
```

Also no need to use the returned `filename`, just upload.